linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -944,6 +949,45 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
       case tree: BinaryOp =>
         pretransformBinaryOp(tree)(cont)
 
+      case WrapAsThrowable(expr) =>
+        pretransformExpr(expr) { texpr =>
+          def default = {
+            val refinedType: RefinedType = RefinedType(ThrowableClassType, isExact = false, isNullable = false)
+            cont(PreTransTree(WrapAsThrowable(finishTransformExpr(texpr)), refinedType))
+          }
+
+          if (isSubtype(texpr.tpe.base, ThrowableClassType)) {
```

Comment:
You mean when it's a nullable `Throwable`, optimize as
```scala
if (texpr !== null) texpr else new js.JavaScriptException(null)
```
?

The problem with that is that we need to repeat `texpr` twice. And therefore create a temporary binding for it. The complexity here doesn't seem to be worth it for what it brings. Does it?

Reply:
Ah, no. Sorry. I meant if `texpr` is a literal `null`, IIUC we can also optimize (because it is UB IIUC).

IIUC we also have this optimization in the compiler.

Reply:
It's not UB to have `null` in either `Wrap` or `Unwrap`. `Wrap(null)` is a `js.JavaScriptException(null)`, and `Unwrap(null)` is `null`.